testing/upload_validation.ts

```diff
@@ -580,8 +580,10 @@ function buildMetadataSchema({sdkVersion}: BuildMetadataSchemaArgs): {
     }),
     [AuthenticationType.OAuth2]: zodCompleteStrictObject<OAuth2Authentication>({
       type: zodDiscriminant(AuthenticationType.OAuth2),
-      authorizationUrl: z.string().url(),
-      tokenUrl: z.string().url(),
+      /** Accepts relative URLs when requiresEndpointUrl is true. */
+      authorizationUrl: z.string(),
+      /** Accepts relative URLs when requiresEndpointUrl is true. */
+      tokenUrl: z.string(),
```

Comment:
Can you do a refine/superRefine so that when requiresEndpoint is false these values are validated as full urls, and when true, validated as relative urls?

Reply:
Done, added some basic validation.